src/Assets/components/CustomTrustline.tsx

```diff
@@ -26,11 +30,37 @@ function CustomTrustlineDialog(props: Props) {
   const [limit, setLimit] = React.useState("")
   const isWidthMax450 = useMediaQuery("(max-width:450px)")
   const { t } = useTranslation()
+  const { openSavedAddresses } = React.useContext(DialogsContext)
 
   const createTransaction = () =>
     props.createAddAssetTransaction(new Asset(code, issuerPublicKey), { limit: limit || undefined })
   const addCustomAsset = () => props.sendTransaction(createTransaction)
 
+  const handleOnSavedAddressClick = React.useCallback(
+    (address: string) => {
+      setIssuerPublicKey(address)
+      openSavedAddresses(null)
+    },
+    [openSavedAddresses]
+  )
+
+  const handleSavedAddressesClick = React.useCallback(() => {
+    openSavedAddresses({
+      onSelect: handleOnSavedAddressClick
+    })
+  }, [handleOnSavedAddressClick, openSavedAddresses])
+
+  const savedAddressesAdornment = React.useMemo(
+    () => (
+      <InputAdornment disableTypography position="end">
+        <Box onClick={handleSavedAddressesClick} style={{ cursor: "pointer" }}>
```

Comment:
Do we need this Box here? Can't we just attach onClick and style to AccountBoxIcon directly? IF Box is needed, explain why?